source/grid/tria.cc

```diff
@@ -11959,9 +11959,6 @@ typename Triangulation<dim, spacedim>::cell_iterator
 Triangulation<dim, spacedim>::last() const
 {
   const unsigned int level = levels.size() - 1;
-
-  Assert(level < n_global_levels() || level < levels.size(),
```

Comment:
could we have a case that `levels.size() - 1 >= n_global_levels()` ?

Reply:
Well, the condition is that some other thing OR `levels<levels.size()` the other thing, but from setting the variable in the line above to `levels.size()-1`, we already know that the second part of the OR condition is true, so the assertion will always succeed.

Reply:
oh, of course! I completely missed OR part...

Reply:
I guess the question is whether `levels.size()` can ever be zero.

Reply:
Only if nothing has ever been put into the triangulation. I have a few follow-up patches in this direction, and should add a different assertion into this function to avoid this case. But that's separate from the question here.

Reply:
Well, the `Assert` triggers if `level.size()` is in fact zero so it is not completely pointless.
If you put something here later, I am fine with merging this PR.